Pear/Profile/MatchProfileViewController.swift

```diff
@@ -0,0 +1,230 @@
+//
+//  MatchProfileViewController.swift
+//  Pear
+//
+//  Created by Lucy Xu on 4/28/21.
+//  Copyright © 2021 cuappdev. All rights reserved.
+//
+
+import GoogleSignIn
+import Kingfisher
+import UIKit
+
+class MatchProfileViewController: UIViewController {
+
+    // MARK: - Private View Vars
+    private let backButton = UIButton()
+    private var chatStatus: ChatStatus?
+    private let match: Match?
+    private var meetupStatusView: MeetupStatusView?
+    private var pair: User?
+    private var profileSections = [ProfileSectionType]()
+    private let profileTableView = UITableView(frame: .zero, style: .plain)
+    private let user: User?
+    private var reachOutButton = UIButton()
+
+    // MARK: - Private Data Vars
+    private let reachOutButtonSize = CGSize(width: 200, height: 50)
+
+    init(match: Match?, user: User?) {
+        self.match = match
+        self.user = user
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.setNavigationBarHidden(true, animated: true)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .backgroundLightGreen
+
+        getPairThen { [weak self] pair in
+            guard let self = self else { return }
+
+            self.pair = pair
+
+            if let match = self.match {
+                self.chatStatus = ChatStatus.forMatch(match: match, pair: pair)
+            }
+
+            self.setupViews(pair: pair)
+            self.setupConstraints()
+
+            self.profileSections = [.summary, .basics]
+            if pair.interests.count > 0 {
+                self.profileSections.append(.interests)
+            }
+            if pair.groups.count > 0 {
+                self.profileSections.append(.groups)
+            }
+            self.profileTableView.reloadData()
+        }
+    }
+
+    private func getPairThen(_ completion: @escaping (User) -> Void) {
```

Comment:
I'm not sure how idiomatic the `then` is but I don't feel strongly about it either way.